apps/studio/components/ui/AIAssistantPanel/AIAssistant.tsx

```diff
@@ -221,6 +226,64 @@ export const AIAssistant = ({ className }: AIAssistantProps) => {
     [snap]
   )
 
+  const editMessage = useCallback(
+    (messageId: string) => {
+      const messageIndex = chatMessages.findIndex((msg) => msg.id === messageId)
+      if (messageIndex === -1) return
+
+      // Target message
+      const messageToEdit = chatMessages[messageIndex]
+
+      // Activate editing mode
+      setEditingMessageId(messageId)
+      setOriginalMessageContent(messageToEdit.content ?? '')
+      setValue(messageToEdit.content ?? '')
+
+      if (inputRef.current) {
+        inputRef.current.focus()
+      }
+    },
+    [chatMessages, setValue]
+  )
+
+  const cancelEdit = useCallback(() => {
+    setEditingMessageId(null)
+    setOriginalMessageContent('')
+    setValue('')
+  }, [setValue])
+
+  const confirmEdit = useCallback(
```

Comment:
do we need an extra function here vs just checking if editingMessageId is set in sendMessageToAssistant and if so deleteMessagesAfter?

Reply:
yea i agree with you: `confirmEdit` and `sendMessageToAssistant` had a lot of duplicated logic.

fixed it! reference: [`af96469` (#37603)](https://github.com/supabase/supabase/pull/37603/commits/af9646995b96df7c104c5077479d064cf4e2a418)